cli/src/base.ts

```diff
@@ -0,0 +1,38 @@
+import { Command } from '@oclif/core';
+import { XataApiClient } from '@xata.io/client';
+import ansiRegex from 'ansi-regex';
+import chalk from 'chalk';
+import dotenv from 'dotenv';
+import fetch from 'node-fetch';
+import table from 'text-table';
+import { readAPIKey } from './key.js';
+
+export abstract class BaseCommand extends Command {
+  locale: string | undefined = undefined;
+  timeZone: string | undefined = undefined;
+
+  async init() {
+    dotenv.config();
+  }
+
+  async getXataClient() {
+    const apiKey = await readAPIKey();
+    if (!apiKey) this.error('Could not instantiate Xata client. No API key found.'); // TODO: give suggested next steps
```

Comment:
I know we partially discussed it last week, and I saw your points on using env variables for API Keys.

BUT... After a second thought, I really think that it would improve the DX of the CLI that if there's no env variable, we prompt the user with a URL to the frontend, that grants a new API Key and the CLI starts a server to receive it as in OAuth. Finally it stores for re-use in an encrypted config.

WDYT? We can probably discuss this in a meeting, and no actions needed in this PR!

Reply:
Edit: Just saw https://github.com/xataio/client-ts/pull/224/files#diff-6a5236bbaf544aebc3a41c104350ad36d5c5a779eeebc40da558de9bbf893f0bR10

Okay, all in now. I think it should still be encrypted and we should offer some managed way to set it up with the frontend but that's the future.

Reply:
Yeah, I'm using the same path than the go xata cli. This is pretty standard: storing a key in a file with read-only permissions for the current user. See https://github.com/xataio/cli/blob/main/config/config.go#L23

I think it's also a good idea to try to read an env variable, specially if you have an `.env` file, to use a key that is scoped to your database potentially and not having to have a broad API key.

I'm open to discussing all this before releasing the first version 👍